cmd/skaffold/app/cmd/parse_config.go

```diff
@@ -184,19 +191,22 @@ func processEachDependency(d latest.ConfigDependency, fPath string, required boo
 
 	if path == "" {
 		// empty path means configs in the same file
-		path = fPath
+		path = cfgOpts.file
 	}
 	fi, err := os.Stat(path)
 	if err != nil {
 		if os.IsNotExist(errors.Unwrap(err)) {
-			return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", path, fPath)
+			return nil, fmt.Errorf("could not find skaffold config %s that is referenced as a dependency in config %s", path, cfgOpts.file)
 		}
-		return nil, fmt.Errorf("parsing dependencies for skaffold config %s: %w", fPath, err)
+		return nil, fmt.Errorf("parsing dependencies for skaffold config %s: %w", cfgOpts.file, err)
 	}
 	if fi.IsDir() {
 		path = filepath.Join(path, "skaffold.yaml")
 	}
-	depConfigs, err := getConfigs(configOpts{file: path, selection: d.Names, profiles: depProfiles, isRequired: required, isDependency: path != fPath}, opts, r)
+	cfgOpts.isDependency = cfgOpts.isDependency || path != cfgOpts.file
```

Comment:
The main change in this PR is:
```diff
- cfgOpts.isDependency = path != cfgOpts.file
+ cfgOpts.isDependency = cfgOpts.isDependency || path != cfgOpts.file
```
Other changes are just reordering functions.

Reply:
`isDependency` defines if the current config is an imported config vs the main config. We want to resolve absolute file paths for properties in imported configs and not on the main config. For multiple configs defined in the same file just checking `path != cfgOpts.file` was causing some imported configs to have `isDependency` as false.

Reply:
thanks for this explanation, makes reviewing a lot easier 👍